src/hoc/withServeryQuery/index.tsx

```diff
@@ -0,0 +1,29 @@
+import { SnapshotEmitter } from '@squidcloud/common';
+import React from 'react';
+import Server from './Server';
+
+export interface WithQueryProps<T> {
+  data: Array<T>;
+}
+
+export const withServerQuery = <C extends React.ComponentType<any>, T>(
```

Comment:
@joey-squid I tried a few different ways of typing this before landing on the solution in the PR. Just tested this one out but I get:

```
TS2322: Type 'Props & { data: DataType[]; }' is not assignable to type 'IntrinsicAttributes & LibraryManagedAttributes<C, Props & WithQueryProps<DataType>>'.
  Type 'Props & { data: DataType[]; }' is not assignable to type 'LibraryManagedAttributes<C, Props & WithQueryProps<DataType>>'.

30     return <Component {...props} data={data} />;
```

Will probably keep it as is for now since it's not exposed to the user, but agree there may be a better way to type this.